quest/quest.js

```diff
@@ -0,0 +1,49 @@
+import loadProfile from '../common/load-profile.js';
+import { getUser, saveUser } from '../data/api.js';
+import quests from '../data/quest-data.js';
+import createChoice from './create-choice.js';
+import findById from '../common/find-by-id.js';
+import scoreQuest from './score-quest.js';
+
+loadProfile();
+
+const searchParams = new URLSearchParams(window.location.search);
+const questId = searchParams.get('id');
+const quest = findById(quests, questId);
+
+if (!quest) {
+    window.location = '../map/index.html';
+}
+
+const title = document.getElementById('title');
+const image = document.getElementById('questImage');
+const description = document.getElementById('description');
+const choiceForm = document.getElementById('choice-form');
+const choices = document.getElementById('choices');
+const result = document.getElementById('result');
+const resultDescription = document.getElementById('result-description');
+
+title.textContent = quest.title;
+image.src = '../assets/quests/' + quest.image;
+description.textContent = quest.description;
+
+for (let i = 0; i < quest.choices.length; i++) {
+    const choice = quest.choices[i];
+    const choiceDOM = createChoice(choice);
+    choices.appendChild(choiceDOM);
+}
+
+choiceForm.addEventListener('submit', function(event) {
+    event.preventDefault();
+    const formData = new FormData(choiceForm);
+    const choiceId = formData.get('choice');
+    const choice = findById(quest.choices, choiceId);
+    const user = getUser();
+    scoreQuest(choice, quest.id, user);
+    saveUser(user);
+
+    choiceForm.classList.add('hidden');
+    result.classList.remove('hidden');
+    resultDescription.textContent = choice.result;
+    loadProfile();
+});
```

Comment:
You're missing logic in this file to return to the map page after the user picks their choice